### PR TITLE
feat: upgrade project to support PHP 8 only

### PR DESCRIPTION
**Pull request by Dobmod**
- Updated README and README_CN
  - Fixed broken links
  - Synced latest content
- Updated composer.json
  - Required PHP 8
  - Synced and upgraded dependencies to the latest versions
  - Removed dependency on `s1lentium/iptools`
- Updated CI configuration
  - Removed tests for PHP 7.x
- Utilized PHP 8 syntax features
  - Nullsafe operator
  - Match expression
  - Replaced functions with `str_contains` and `str_starts_with`
  - Used union types for better type handling

### CODE REVIEW

File: src/Util/BuiltinOperations.php [hunk 1]
```diff
@@ -364,11 +364,18 @@ public static function regexMatchFunc(...$args): bool
...
+            $ip = ip2long($ip1);
+            $subnet = ip2long($subnet);
+            $mask = 0xffffffff << (32 - intval($mask));
 
-        return $objIP2->contains($objIP1);
+            return ($ip & $mask) === ($subnet & $mask);
+        } else {
+
+            return $ip1 === $ip2;
+        }
```
Comment (leeqvip) [inline src/Util/BuiltinOperations.php:378]: Can it work with ipv6 ?

Comment (Dobmod) [reply src/Util/BuiltinOperations.php]: IPv6 support and addtional unit tests have been added this time.

File: src/Model/Assertion.php [hunk 2]
```diff
@@ -74,26 +73,20 @@ public function buildRoleLinks(RoleManager $rm): void
...
-                $this->rm->addLink($rule[0], $rule[1], $rule[2]);
-            } elseif (4 == $count) {
-                $this->rm->addLink($rule[0], $rule[1], $rule[2], $rule[3]);
+            if(\count($rule) > $count) {
+                $rule = \array_slice($rule, 0, $count);
             }
-        }
 
-        Log::logPrint('Role links for: ' . $this->key);
-        $this->rm->printRoles();
```
Comment (leeqvip) [inline src/Model/Assertion.php:96]: This shouldn't be removed.

File: src/Util/BuiltinOperations.php [hunk 3]
```diff
@@ -364,11 +364,38 @@ public static function regexMatchFunc(...$args): bool
...
+                $subnet = ip2long($subnet);
+                $mask = 0xffffffff << (32 - $prefixLength);
+
+                return ($ip & $mask) === ($subnet & $mask);
+            } else {
+                return false;
+            }
+        } else {
+            return $ip1 === $ip2;
+        }
```
Comment (leeqvip) [inline src/Util/BuiltinOperations.php:398]: Please move this part to the `Util.php`.

**Review by leeqvip — CHANGES_REQUESTED** [review]

File: src/Model/Assertion.php (hunk 2 above)
Comment (Dobmod) [reply src/Model/Assertion.php:96]: Really? This code has been removed from the Core Library.

**Pull request merged**